src/Blazor.FlexGrid/DataSet/TableDataSet.cs

```diff
@@ -132,7 +141,83 @@ public void CancelEditation()
         private void LoadFromQueryableSource()
         {
             PageableOptions.TotalItemsCount = ApplyDeletedConditionToQueryable(source).Count();
-            Items = ApplyFiltersToQueryable(source).ToList();
+            if (!GroupingOptions.IsGroupingActive)
+            {
+                Items = ApplyFiltersToQueryable(source).ToList();
+            }
+            else
+            {
+                GroupedItems = ApplyFiltersWithGroupingToQueryable(source);
+            }
+        }
+
+        private IQueryable<GroupItem<TItem>> ApplyFiltersWithGroupingToQueryable(IQueryable<TItem> source)
+        {
+            try
+            {
+                var groupedItems = this.GroupItems<TItem>(source);
+                groupedItems = ApplyPagingToGroupedQueryable(groupedItems);
+                groupedItems = ApplySortingToGroupedQueryable(groupedItems);
+
+
+                if (this.GroupedItems == null)
+                {
+
+                    return groupedItems.Select(item => new GroupItem<TItem>(item.Key, item.Items) { IsCollapsed = true });
```

Comment:
Move set of `IsCollapsed` into constructor of `GroupItem` or make defaul value `true` and this can be deleted.